vitals/battery/LiIonBattery.java

```diff
@@ -0,0 +1,187 @@
+package vitals.battery;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import vitals.TemperatureScale;
+import vitals.ValidateBMSVariants;
+import vitals.bms.BMS;
+import vitals.language.Language;
+
+import static vitals.bms.BMSCheck.*;
+
+/**
+ * @author {@literal Jayaram Naveenkumar (jayaram.naveenkumar@in.bosch.com)}
+ */
+public class LiIonBattery {
+
+    static final BiFunction<Float, Float, Float> calculatePercentageValue =
+          (percentageValue, value) -> (percentageValue / 100) * value;
+    private Map<BMS, Float> input;
+
+    public LiIonBattery(
+          float temperature,
+          float soc,
+          float chargeRate
+    )
+    {
+        saveInput(temperature, soc, chargeRate);
+    }
+
+    private void saveInput(
+          float temperature,
+          float soc,
+          float chargeRate
+    )
+    {
+        input = new TreeMap<>();
+        input.put(BMS.TEMPERATURE, temperature);
+        input.put(BMS.SOC, soc);
+        input.put(BMS.CHARGINGRATE, chargeRate);
+    }
+
+    private List<ValidateBMSVariants> variants() {
```

Comment:
Earlier variants function was simple .... as new requirements are added ...construction of new ValidateBMSVariants need to be changed....why don't you separate construction of ValidateBMSVariants from LionBattery Class and parameterize it

Reply:
as I couldn't come up with a solution, can u provide an idea on how to parameterize it